builtin/providers/aws/json.go

```diff
@@ -0,0 +1,38 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+func normalizeJson(jsonString interface{}) string {
+	if jsonString == nil || jsonString == "" {
+		return ""
+	}
+	var j interface{}
+	err := json.Unmarshal([]byte(jsonString.(string)), &j)
+	if err != nil {
+		return fmt.Sprintf("Error parsing JSON: %s", err)
+	}
+	b, err := json.Marshal(j)
+	if err != nil {
+		panic(err) // unexpected
+	}
+	return string(b)
+}
+
+func normalizePolicyDocument(policyString interface{}) string {
+	if policyString == nil || policyString == "" {
+		return ""
+	}
+	var policy IAMPolicyDoc
+	err := json.Unmarshal([]byte(policyString.(string)), &policy)
+	if err != nil {
+		return fmt.Sprintf("Error parsing IAM policy document: %s", err)
```

Comment:
This doesn't feel right... seems like if an error occurs here we'd end up storing this error message in the state, which I don't think is what you intended here.

Probably the best we could do here is to make this function silently change nothing in the event of an error (just return the provided string verbatim) and trust the backend to catch and return any syntax errors. I suppose we could couple this with a policy validation function that we could use as `ValidateFunc` on these attributes and thus catch certain errors locally during Terraform validation, and thus assume that the policy will always be valid by the time it gets in here.

Also wondering about the rationale for the `policyString` argument being `interface{}` rather than `string` here... it seems like this code would panic at runtime if the policy were anything other than a string, so could we just make the argument be of type `string` in the first place?